packages/cli/src/models/files/ZosNetworkFile.ts

```diff
@@ -0,0 +1,365 @@
+      deployedBytecodeHash: bytecodeDigest(instance.constructor.binary)
+    };
+  }
+
+  public unsetSolidityLib(libName: string): void {
+    delete this.data.solidityLibs[libName];
+  }
+
+  public setSolidityLib(alias: string, value: any): void {
+    if (!this.data.solidityLibs) this.data.solidityLibs = {};
+    this.data.solidityLibs[alias] = value;
+  }
+
+  public solidityLib(libName: string): any {
+    return this.data.solidityLibs[libName];
+  }
+
+  public getSolidityLibs(libs: any): any {
+    const { solidityLibs } = this.data;
+
+    return Object
+      .keys(solidityLibs)
+      .filter((libName) => libs.includes(libName))
+      .map((libName) => ({ libName, address: solidityLibs[libName].address }))
+      .reduce((someLibs, currentLib) => {
+        someLibs[currentLib.libName] = currentLib.address;
+        return someLibs;
+      }, {});
+  }
+
+  public hasSolidityLib(libName: string): boolean {
+    return !_.isEmpty(this.solidityLib(libName));
+  }
+
+  public solidityLibsMissing(libs: any): string[] {
+    return _.difference(Object.keys(this.solidityLibs), libs);
+  }
+
+  public getSolidityLibOrContract(aliasOrName: string): any {
+    return this.data.solidityLibs[aliasOrName] || this.data.contracts[aliasOrName];
+  }
+
+  public hasSolidityLibOrContract(aliasOrName: string): boolean {
+    return this.hasSolidityLib(aliasOrName) || this.hasContract(aliasOrName);
+  }
+
+  public updateImplementation(aliasOrName: string, fn: (...args: any[]) => any): void {
+    if (this.hasContract(aliasOrName)) this.data.contracts[aliasOrName] = fn(this.data.contracts[aliasOrName]);
+    else if (this.hasSolidityLib(aliasOrName)) this.data.solidityLibs[aliasOrName] = fn(this.data.solidityLibs[aliasOrName]);
+    else return;
+  }
+
+  get dependencies(): any[] {
+    return this.data.dependencies || {};
+  }
+
+  get dependenciesNames(): string[] {
+    return Object.keys(this.dependencies);
+  }
+
+  public getDependency(name: string): any | null {
+    if (!this.data.dependencies) return null;
+    return this.data.dependencies[name] || {};
+  }
+
+  public hasDependency(name: string): boolean {
+    return !_.isEmpty(this.getDependency(name));
+  }
+
+  public hasDependencies(): boolean {
+    return !_.isEmpty(this.dependencies);
+  }
+
+  public getProxies({ package: packageName, contract, address }: {package?: string, contract?: any, address?: string} = {}): any[] {
+    if (_.isEmpty(this.data.proxies)) return [];
+    const allProxies = _.flatMap(this.data.proxies || {}, (proxiesList, fullname) => (
+      _.map(proxiesList, (proxyInfo) => ({
+        ...fromContractFullName(fullname),
+        ...proxyInfo
+      }))
+    ));
+    return _.filter(allProxies, (proxy) => (
+      (!packageName || proxy.package === packageName) &&
+      (!contract || proxy.contract === contract) &&
+      (!address || proxy.address === address)
+    ));
+  }
+
+  // TS-TODO: return type can probably be typed to Proxy.ts from lib?
```

Comment:
`ZosNetworkFile` edits/inserts/deletes `zos.network.json` content, and doesn't manage any other class instance such as `Proxy` from lib. 
Here comes the last pair of interfaces I would define: `ProxyInterface` and `ProxiesInterface`.